### PR TITLE
EQ FIR optimize proposal

### PR DESCRIPTION
**Pull request by singalsu**
This patch optimizes FIR EQ computation speed a lot when compiled with xt-xcc.

The IPC is updated to match ALSA conventions with binary and enum controls.

Also code style is fixed to pass the pre-commit and post-commit checks.

### CODE REVIEW

File: src/audio/fir_config.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+#if FIR_AUTOARCH == 0
+#define FIR_GENERIC	0
+#define FIR_HIFIEP	0
+#define FIR_HIFI3	1
+#endif
+
+/* Select optimized code variant when xt-xcc compiler is used */
+#if FIR_AUTOARCH == 1
+#if defined __XCC__
+//#include <xtensa/config/core-isa.h>
```
Comment (lgirdwood) [inline src/audio/fir_config.h]: needed ?

Comment (singalsu) [reply src/audio/fir_config.h]: I changed the pull request into next from edit menu but there's a lot of other commits and even conflicts in other parts than my work. It looks like next was behind master.

Comment (lgirdwood) [reply src/audio/fir_config.h]: @singalsu you will need to rebase on top of next, this will remove other commits

**Comment by singalsu** [comment]
@lgirdwood I've now rebased to next and fixed that // comment line. It was a mistake and that header file that I had commented out for some weird reason needs to be included.

**Pull request merged**